### DISCUSSION TITLE
HDD/SSD interface incorrect. Windows limitation?

### DISCUSSION THREAD

**Discussion by ninjamonkey198206** [amidaware/tacticalrmm#1080] [Q&A] [answered]
I've noticed that agents I have running display either IDE (SATA devices) or SCSI (NVMe), when they are obviously not the reported interface type.

Is this an issue with what Windows reports, or the location the data is being pulled from in Windows?

Currently running 0.12.4 via Docker.

It is a Windows specific issue, as the Linux agent seems to report somewhat correctly. The only issue being SATA drives connected via an SAS HBA with breakout cables. Those also get reported as SCSI, which I assume is due to the controller/HBA.

**Comment by wh1te909** [comment, marked as answer]
the agent uses windows api to get disk information, specifically this https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-diskdrive

so it's an issue with how windows reports the drive type

**Reply by ninjamonkey198206** [reply]
Gotcha. No worries, then. It's fairly easy to tell which is which. I'll go bark at the Microsoft folks I know and give them a hard time about it. Don't think they have anything to do with it, but I like to give 'em shit whenever I can about Windows.
